framework/src/wifi_manager/slsi_wifi_utils_lwnl80211.c

```diff
@@ -0,0 +1,483 @@
+		WU_LOG(WU_TAG"[ERR:%s] %s %s:%d code(%s)\n",                            \
+			   WU_TAG, __FUNCTION__, __FILE__, __LINE__, strerror(errno));      \
+	} while (0)
+
+#define WU_ERR_FD(fd)                                                           \
+	do {                                                                        \
+		WU_LOG(WU_TAG"[ERR:%s] %s %s:%d fd(%d) code(%s)\n",                     \
+			   WU_TAG, __FUNCTION__, __FILE__, __LINE__, fd, strerror(errno));  \
+	} while (0)
+
+#define WU_CHECK_ERR(arg)                                    \
+	do {                                                     \
+		if (arg < 0) {                                       \
+			WU_ERR;                                          \
+			return WIFI_UTILS_FAIL;                          \
+		}                                                    \
+	} while (0)
+
+#define WU_ENTER                                                                \
+	do {                                                                        \
+		WU_LOG(WU_TAG"%s\t%s:%d\n", __FUNCTION__, __FILE__, __LINE__);          \
+	} while (0)
+
+#define WU_CALL(fd, hnd, code, param)                                           \
+	do {                                                                        \
+		int res = ioctl(fd, code, (unsigned long)((uintptr_t)&param));          \
+		if (res < 0) {                                                          \
+			WU_CLOSE(fd);                                                       \
+			WU_ERR_FD(fd);                                                      \
+			return WIFI_UTILS_FAIL;                                             \
+		}                                                                       \
+	} while (0)
+
+#define WU_CALL_ERROUT(fd, hnd, code, param)                                    \
+	do {                                                                        \
+		int res = ioctl(fd, code, (unsigned long)((uintptr_t)&param));          \
+		if (res < 0) {                                                          \
+			WU_ERR;                                                             \
+			ret = WIFI_UTILS_FAIL;                                              \
+			goto errout;                                                        \
+		}                                                                       \
+	} while (0)
+
+#define WU_CLOSE(fd)                            \
+	do {                                        \
+		close(fd);                              \
+	} while (0)
+
+static wifi_utils_cb_s g_cbk = {NULL, NULL, NULL, NULL, NULL};
+
+struct _wifi_utils_s g_hnd = {"", 0, 0};
+
+static void free_scan_data(wifi_utils_scan_list_s *scan_list)
+{
+	wifi_utils_scan_list_s *cur = scan_list;
+	wifi_utils_scan_list_s *prev = NULL;
+	while (cur) {
+		prev = cur;
+		cur = cur->next;
+		free(prev);
+	}
+	scan_list = NULL;
+}
+
+static wifi_utils_result_e receive_scan_data(mqd_t mq, wifi_utils_scan_list_s *scan_list)
```

Comment:
Please provide function description and details about input and return arguments.

Reply:
This function is static, locally-used internally. Thanks.